services/src/main/java/org/keycloak/services/resources/admin/AdminConsole.java

```diff
@@ -296,10 +297,38 @@ public Response getMainPage() throws URISyntaxException, IOException, FreeMarker
         }
     }
 
+    private Theme getTheme() throws IOException {
+        ThemeProvider themeProvider = session.getProvider(ThemeProvider.class, "extending");
+        return themeProvider.getTheme(realm.getAdminTheme(), Theme.Type.ADMIN);
+    }
+
     @GET
     @Path("{indexhtml: index.html}") // this expression is a hack to get around jaxdoclet generation bug.  Doesn't like index.html
     public Response getIndexHtmlRedirect() {
         return Response.status(302).location(uriInfo.getRequestUriBuilder().path("../").build()).build();
     }
 
+    @GET
+    @Path("messages.json")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Properties getMessages(@QueryParam("lang") String lang) {
+        if (lang == null) {
+            logger.warn("Locale not specified for messages.json");
+            lang = "en";
+        }
+
+        try {
+            Properties msgs = AdminMessagesLoader.getMessages(getTheme(), lang);
+            if (msgs.isEmpty()) {
+                logger.warn("Message bundle not found for language code '" + lang + "'");
+                msgs = AdminMessagesLoader.getMessages(getTheme(), "en"); // fall back to en
```

Comment:
Looking at this closer, the default locale will work as expected.  The cookie has to be set at login, so the admin UI will act accordingly.  

The two situations in lines 317 and 324 should really never happen.  So if they do happen then there is some other problem that makes the default value likely to fail was well.  It has to be set to something, so I went with "en".